File: src/main/java/me/kuku/yuq/entity/StepEntity.java
package me.kuku.yuq.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table(name = "motion")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class StepEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(unique = true)
    private Long qq;
    @Column(name = "group_")
    private Long group;
    private String leXinPhone;
    private String leXinPassword;
    @Lob
    @Column(columnDefinition="text")
    private String leXinCookie;
    private String leXinUserId;
    @Lob
    @Column(columnDefinition="text")
    private String leXinAccessToken;
    private Boolean leXinStatus;
    private Integer step;
    private String miPhone;
    private String miPassword;
    @Lob
    @Column(columnDefinition="text")
    private String miLoginToken;
    private Boolean miStatus;

    public StepEntity(String leXinPhone, String leXinPassword, String leXinCookie, String leXinUserId, String leXinAccessToken){
        this.leXinPhone = leXinPhone;
        this.leXinPassword = leXinPassword;
        this.leXinCookie = leXinCookie;
        this.leXinUserId = leXinUserId;
        this.leXinAccessToken = leXinAccessToken;
    }

    public StepEntity(Long qq, Long group){
        this.group = group;
        this.leXinStatus = false;
        this.miStatus = false;
        this.qq = qq;
        this.step = 0;
    }

    public StepEntity(String miPhone, String miPassword, String miLoginToken){
        this.miPhone = miPhone;
        this.miPassword = miPassword;
        this.miLoginToken = miLoginToken;
    }
}

File: src/main/kotlin/me/kuku/yuq/entity/StepEntity.kt
package me.kuku.yuq.entity

import org.hibernate.annotations.Type
import org.springframework.data.jpa.repository.JpaRepository
import javax.inject.Inject
import javax.persistence.*

@Entity
@Table(name = "step")
class StepEntity: BaseEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null
    @OneToOne
    @JoinColumn(name = "qq_id")
    var qqEntity: QqEntity? = null
    @Column(length = 5000)
    var leXinCookie: String = ""
    var leXinUserid: String = ""
    @Column(length = 5000)
    var leXinAccessToken: String = ""
    @Column(length = 1000)
    var miLoginToken: String = ""
    @Type(type = "json")
    @Column(columnDefinition = "json")
    var config: StepConfig = StepConfig()
}

interface StepRepository: JpaRepository<StepEntity, Int> {
    fun findByQqEntity(qqEntity: QqEntity): StepEntity?
}

class StepService @Inject constructor(
    private val stepRepository: StepRepository
) {

    fun save(stepEntity: StepEntity): StepEntity = stepRepository.save(stepEntity)

    fun findAll(): List<StepEntity> = stepRepository.findAll()

    fun findByQqEntity(qqEntity: QqEntity) = stepRepository.findByQqEntity(qqEntity)
}

class StepConfig(var step: Int = -1)